pkg/daemon/rpm-ostree.go

```diff
@@ -307,6 +321,55 @@ func (r *RpmOstreeClient) Rebase(imgURL, osImageContentDir string) (changed bool
 	return
 }
 
+// Rebase potentially rebases system if not already rebased.
+func (r *RpmOstreeClient) RebaseLayered(imgURL string) (changed bool, err error) {
+
+	defaultDeployment, err := r.GetBootedDeployment()
+	if err != nil {
+		return
+	}
+
+	previousPivot := ""
+	if len(defaultDeployment.CustomOrigin) > 0 {
+		if strings.HasPrefix(defaultDeployment.CustomOrigin[0], "pivot://") {
+			previousPivot = defaultDeployment.CustomOrigin[0][len("pivot://"):]
+			glog.Infof("Previous pivot: %s", previousPivot)
+		} else {
+			glog.Infof("Previous custom origin: %s", defaultDeployment.CustomOrigin[0])
+		}
+	} else {
+		glog.Info("Current origin is not custom")
+	}
+
+	// This will be what will be displayed in `rpm-ostree status` as the "origin spec"
+	customURL := fmt.Sprintf("pivot://%s", imgURL)
+	glog.Infof("Executing rebase to %s", imgURL)
+
+	args := []string{"rebase", "--experimental", "ostree-unverified-registry:" + imgURL,
+		"--custom-origin-url", customURL, "--custom-origin-description", "Layered image managed by machine-config-operator"}
```

Comment:
I don't think we need the custom origin stuff anymore.  IMO, that was a way for us to describe "out of band" the container image information before rpm-ostree natively understood it.